AlphaWallet/Settings/Models/AddCustomChain.swift

```diff
@@ -65,11 +72,14 @@ class AddCustomChain {
         }.done {
             self.delegate?.notifyAddCustomChainQueuedSuccessfully(in: self)
         }.catch {
-            if let error = $0 as? AddCustomChainError {
-                self.delegate?.notifyAddCustomChainFailed(error: error, in: self)
-            } else if case PMKError.cancelled = $0 {
-                //no-op
-            } else {
+            switch $0 {
+            case SessionTaskError.connectionError:
+                self.delegate?.notifyRpcURlHostnameFailure()
+            case is AddCustomChainError:
+                self.delegate?.notifyAddCustomChainFailed(error: ($0 as! AddCustomChainError), in: self)
+            case PMKError.cancelled:
+                return
+            default:
                 self.delegate?.notifyAddCustomChainFailed(error: .others("\(R.string.localizable.addCustomChainErrorUnknown()) — \($0)"), in: self)
```

Comment:
We can leave it, but generally, I'd prefer if we reserve `switch-default:` for enums and almost never use it (so that we get compilation errors when we introduce new enum values). It's much easier to discover them if we limit them to enums.